packages/dev/core/src/Collisions/gpuPicker.ts

```diff
@@ -176,14 +176,14 @@ export class GPUPicker {
 
         const effect = material.getEffect()!;
 
-        if (!mesh.hasInstances && !mesh.isAnInstance && !mesh.hasThinInstances) {
+        if (!mesh.hasInstances && !mesh.isAnInstance && !mesh.hasThinInstances && typeof this._idColors[mesh.uniqueId] !== "undefined") {

```

Comment:
why do you need the typeof ? This is something we try to avoid for perf reasons

Reply:
ok, fixed!

Reply:
isn't `!this._idColors[mesh.uniqueId]` enough here?